cowprotocol/cowswap
# Keep track of in-flight tx and avoid order Id collisions

# Summary

Continues #1741
Closes #1301

Ready for review although haven't tested, I will document more the PR (but I need to go to the Christmas dinner :P ) 
I will leave in draft until I manage to add some description

Basically it deals with collisions of order Id for ETH FLow

- Keeps track of in-flight orders for ETH Flow
- Once they are mined, then we remove them from the in-flight
- The algorithm will keep increasing the fee by one wei recursively until we find no collisions with order ids


# To Test
- Try to quickly create 2 ETH FLOW orders
- Verify the two have a different order ID
- Both should execute
- Repeat the test for "expert mode"


If you want to double check how the fee is bumped, you can check the logs

<img width="1539" alt="image" src="https://user-images.githubusercontent.com/2352112/208178429-3a96894c-6767-4f78-ba60-bc768b22b2fe.png">



If you observe the two consecutive orders in the explorer, they should have one wei difference in the fee
